### PR TITLE
refactor(e2e): embed `bats-support` as git submodule

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Instead of depending on locally installed (and outdated) `bats-support`, we can now embed it as a git submodule ([which is the currently recommended way of installing bats](https://bats-core.readthedocs.io/en/stable/tutorial.html#quick-installation)).
This removes the need to `export BATSLIB=...` and installing it in the first place.
In order to take advantage of new setup, `sdk` repository needs to be pulled with submodules:
- for already cloned repositories:
   ```bash
   cd sdk
   git submodule update --init --recursive
   ```
- when cloning fresh:
   ```bash
   git clone --recursive git://github.com/dfinity/sdk.git
   ``` 

# How Has This Been Tested?

Covered by CI

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly. 
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

**Comment by krpeacock** [comment]
The tests are still failing to run for me locally, but bats works like a charm now!

image

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -48,7 +44,11 @@ If there is already an entry in the 'Unreleased' section, change it; if not, add
     export utils="$(pwd)/e2e/utils"
     export assets="$(pwd)/e2e/assets"
     ```
-1. Install `jq`.
+1. Install `jq` and `sponge`.
+1. Download all of this repository' submodules (which includes `bats-suppor`)
```
Comment (ericswanson-dfinity) [inline .github/CONTRIBUTING.md:48]:
```suggestion
1. Download all of this repository' submodules (which includes `bats-support`)
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]
Nice!

**Review by ericswanson-dfinity — APPROVED** [review]
Nice!

**Pull request merged**